File: practice_programs/write_json.py
import json

# Function to write data to a JSON file
def write_json_file(file_path, data):
    try:
        with open(file_path, 'w') as file:
            json.dump(data, file, indent=4)  # Write JSON data with pretty formatting
            print(f"Data successfully written to {file_path}")
    except Exception as e:
        print(f"An error occurred: {e}")

# Example usage
if __name__ == "__main__":
    file_path = "output.json"
    data = {
        "name": "Alice",
        "age": 25,
        "skills": ["Python", "Data Analysis", "Machine Learning"]
    }
    write_json_file(file_path, data)

File: java_practice programs/write_json.java
import org.json.JSONObject;
import java.nio.file.Files;
import java.nio.file.Paths;
import java.util.HashMap;
import java.util.Map;

public class WriteJsonFile {
    public static void main(String[] args) {
        String filePath = "output.json";

        // Create a map with data
        Map<String, Object> data = new HashMap<>();
        data.put("name", "Alice");
        data.put("age", 25);
        data.put("skills", new String[]{"Java", "Spring", "Hibernate"});

        // Convert the map to a JSON object
        JSONObject json = new JSONObject(data);

        try {
            // Write the JSON object to a file
            Files.write(Paths.get(filePath), json.toString(4).getBytes());
            System.out.println("Data successfully written to " + filePath);
        } catch (Exception e) {
            System.out.println("An error occurred: " + e.getMessage());
        }
    }
}